.github/workflows/build_and_test.yaml

```diff
@@ -70,3 +74,33 @@ jobs:
       if: github.event_name == 'push' && github.ref == 'refs/heads/main'
       # tag is set to `latest` when pushing to main branch
       run: make image.push.multiarch TAG=latest PLATFORMS="linux_amd64 linux_arm64" IMAGE=envoyproxy/gateway-dev
+
+  canary-release:
```

Comment:
can you move this to a pre-release.yaml file mimicking https://github.com/envoyproxy/gateway/blob/main/.github/workflows/release.yaml

Reply:
I think this is a part a build-and-test, and this requires build and tests passed correctly, moving this to another workflow makes it hard to know if build and test passed.

Reply:
you can use the merge to main trigger to trigger the GHA, if something has been merged, it must have built and passed successfully

Reply:
Yes, I think this makes sense, I will separate them.